portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -88,4 +99,48 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     // Redirect user back to the comments page.
     response.sendRedirect("/comments.html");
   }
+
+  /**
+   * Converts the messages ArrayList into a JSON string using manual String concatentation.
+   */
+  private String convertToJson(List<String> comments, List<String> emails) {
```

Comment:
I forget, is this a reversion of the gson stuff from b4?

Why not just make a map and gson that?

Reply:
Yeah, I revert back to using manual string building. The way I build my Json manually here makes it so that I can the comments (and their corresponding emails) are put in order of how when they were made (oldest first, and then I can loop from the beginning of the arrays). I though about using a map, but I assumed it would throw off the order of when the comments were made. Is this not the case?

Reply:
hmmm why does gson mess with the ordering? If it's unavoidable, perhaps you could add a timestamp to the list items and sort client-side?

Reply:
I actually don't know if it messes with the ordering, but I'll try it out and see what happens. (I honestly was just assuming based on my understanding of how  Hashmaps works, hashing into pretty random slots in an array essentially. That's why I though things would get messed up)

Reply:
aaaaaahhhh I see now. Okay, I meant that if you were trying to return both lists like:

```
{
  'foo': [1, 2, 3],
  'bar': [x, y, z]
}
```

I bet you could use the map for that?

Reply:
Ohhh I didn't even think to do it like that! I'll definitely try that!